src/app/clusters/temperature-measurement-server/temperature-measurement-server.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/****************************************************************************
+ * @file
+ * @brief Implementation for the Temperature Measurement Server Cluster
+ ***************************************************************************/
+
+#include <app-common/zap-generated/af-structs.h>
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/util/attribute-storage.h>
+#include <lib/core/CHIPEncoding.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/PlatformManager.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::TemperatureMeasurement;
+using namespace chip::app::Clusters::TemperatureMeasurement::Attributes;
+
+constexpr int16_t kMinMeasuredValueLowerLimit = -27315; // -273.15°C
+constexpr int16_t kMinMeasuredValueUpperLimit = 32766;  // 327.66°C
+constexpr int16_t kMaxMeasuredValueLowerLimit = -27314; // -273.14°C
+constexpr int16_t kMaxMeasuredValueUpperLimit = 32767;  // 327.67°C
+constexpr int16_t kMaxToleranceValue          = 2048;
+
+// =============================================================================
+// Pre-change callbacks for cluster attributes
+// =============================================================================
+
+using Status = Protocols::InteractionModel::Status;
+
+Protocols::InteractionModel::Status MatterTemperatureMeasurementClusterServerPreAttributeChangedCallback(
```

Comment:
I am wondering why this is here at all.  The attributes are not writable.  Are we trying to protect the app author from accidentally setting an invalid value?  But this is also forcing them to set min/max/measured in the right order, which is not documented anywhere....